### ISSUE TITLE
Core file generation doesn't work on macOS 12.6 ARM64

### ISSUE THREAD

**Issue by XVilka** [rizinorg/rizin#3054]
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | macOS 12.6 ARM64 (M1)
| File format of the file you reverse (mandatory)      | MachO
| Architecture/bits of the file (mandatory)            | ARMv8 64
| `rizin -v` full output, **not truncated** (mandatory)  | rizin 0.5.0 @ darwin-arm-64
commit: 7c0bd7cd036d6500bb2ef29adf3d50f0529786a8

### Expected behavior

```
rizin -d $(which rz-ax)
[0x1045d4560]> dg
WARNING: core: Writing to file 'core.93555'
[0x1045d4560]>
```
Core file should be written and be loadable in Rizin afterwards

### Actual behavior
```
rizin -d $(which rz-ax)
[0x1045d4560]> dg
WARNING: core: Writing to file 'core.93555'
^C
```
Even <Ctrl-C> doesn't work, only `kill -9 <rizin pid>`

Generate core file successfully, which can be loaded 

It's implemented in "native" debugger plugin, see these places:
- `librz/debug/p/debug_native.c` in `rz_debug_gcore()` function
- `librz/debug/p/native/xnu/xnu_debug.c` in `xnu_generate_corefile()` function

**Comment by marcograss** [comment]
OS/arch/bits (mandatory) | macOS 12.6 x64
-- | --
File format of the file you reverse (mandatory) | MachO
Architecture/bits of the file (mandatory) | x64
rizin -v full output, not truncated (mandatory) | rizin 0.5.0 @ darwin-x86-64
commit: 7c0bd7cd036d6500bb2ef29adf3d50f0529786a8

```
rizin -d $(which rz-ax)
 -- Add comments using the ';' key in visual mode or the 'CC' command from the rizin shell
[0x115cd3ae0]> dg
WARNING: core: Writing to file 'core.48978'
^C[1]    48977 killed     rizin -d $(which rz-ax)
```
it's stuck had to kill it, the generated core file is empty 0 bytes.

**Comment by thestr4ng3r** [comment]
Strange, it "works" for me on both x86 and arm64. (Meaning it writes out some broken mach-o file at least)
Could you please attach a debugger to it when it is stuck and paste a stacktrace?

**Comment by marcograss** [comment]
Do you have system integrity protection disabled?

**Comment by thestr4ng3r** [comment]
Good point, yes it is currently disabled here at least on the arm64 machine.

Though this does not seem to be the relevant point here, after enabling it still goes through:
![Bildschirmfoto 2022-10-01 um 10 11 37](https://user-images.githubusercontent.com/1460997/193401577-ffa2b78a-682e-4401-9c67-eeb022e92be0.png)

**Comment by marcograss** [comment]
I just tried another x64 machine on 12.6, it doesn't work there also, on commit 1ed4a3aee56a3af96b68ed08a12923769f8982c6
